x-pack/plugin/spatial/src/internalClusterTest/java/org/elasticsearch/xpack/spatial/SpatialDiskUsageIT.java

```diff
@@ -0,0 +1,152 @@
+        ).actionGet();
+
+        XContentBuilder builder = XContentBuilder.builder(XContentType.JSON.xContent());
+        resp.toXContent(builder, ToXContent.EMPTY_PARAMS);
+        XContentParser parser = XContentType.JSON.xContent()
+            .createParser(XContentParserConfiguration.EMPTY.withRegistry(xContentRegistry()), BytesReference.bytes(builder).array());
+        Map<String, Object> objects = parser.map();
+        assertNotNull(objects);
+
+        int value = extractValue(new String[] { "test-index", "store_size_in_bytes" }, objects);
+        assertThat(value, greaterThan(100));
+
+        value = extractValue(new String[] { "test-index", "fields", "location", "total_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+
+        value = extractValue(new String[] { "test-index", "fields", "location", "points_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+
+        value = extractValue(new String[] { "test-index", "fields", "_source", "inverted_index", "total_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_source", "stored_fields_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+        value = extractValue(new String[] { "test-index", "fields", "_source", "points_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_source", "doc_values_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+
+        value = extractValue(new String[] { "test-index", "fields", "_id", "inverted_index", "total_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+        value = extractValue(new String[] { "test-index", "fields", "_id", "stored_fields_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+        value = extractValue(new String[] { "test-index", "fields", "_id", "points_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_id", "doc_values_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "inverted_index", "total_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "stored_fields_in_bytes" }, objects);
+        assertThat(value, equalTo(0));
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "points_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+        value = extractValue(new String[] { "test-index", "fields", "_seq_no", "doc_values_in_bytes" }, objects);
+        assertThat(value, greaterThan(0));
+    }
+
+    @SuppressWarnings("unchecked")
+    private int extractValue(String[] path, Map<String, Object> objects) {
+        for (int i = 0; i < path.length - 1; i++) {
```

Comment:
Can we delegate these to `XContentMapValues#extractValue` instead?

Reply:
oh yes, thanks!